geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -598,7 +599,9 @@
       StatisticsFactory statisticsFactory =
           internalCache.getInternalDistributedSystem().getStatisticsManager();
       if (isGatewayReceiver()) {
-        stats = GatewayReceiverStats.createGatewayReceiverStats(statisticsFactory, sockName);
+        MeterRegistry meterRegistry = internalCache.getMeterRegistry();
+        stats = GatewayReceiverStats.createGatewayReceiverStats(statisticsFactory, sockName,
```

Comment:
Considering that the `GatewayReceiverStats` class already knows how the instrumentation should be done, have you though about passing the `InternalCache` as a parameter to the `createGatewayReceiverStats` method instead of `MeterRegistry`?.
I'm just worried about introducing a new instrumentation dependency within the class itself here, but maybe I'm missing something.